com.unity.netcode.gameobjects/Runtime/Core/NetworkBehaviourUpdater.cs

```diff
@@ -8,12 +8,17 @@ namespace Unity.Netcode
     /// </summary>
     public class NetworkBehaviourUpdater
     {
-        private HashSet<NetworkObject> m_Touched = new HashSet<NetworkObject>();
+        private HashSet<NetworkObject> m_DirtyNetworkObjects = new HashSet<NetworkObject>();
```

Comment:
It's worth considering refactoring this to use NetworkBehaviour rather than NetworkObject, as it allows CouldHaveDirtyNetworkVariables() to be removed and potentially improves perf a bit.


See https://github.com/Unity-Technologies/com.unity.netcode.gameobjects/compare/develop...davidvogelunity:com.unity.netcode.gameobjects:performance#diff-085339e5943eadcc143e946e65a2296d319b41e60482faab74e76ff83a28ce73R8 for how I did it

Reply:
That is an interesting suggestion... 👍

Reply:
This one is a bit out of scope at the moment. Let's grab the perf gains we can here, and focus on reducing our issues backlog